src/pages/Admin/AdminEditComponent_v2.tsx

```diff
@@ -0,0 +1,545 @@
+  const venue = venues?.find((venue) => venue.id === venueId);
+  const unplacedVenues = useMemo(() => venues?.filter(isUnplaced), [venues]);
+  const selfPlacedVenues = useMemo(() => venues?.filter(isSelfPlaced), [
+    venues,
+  ]);
+  const placedVenues = useMemo(() => venues?.filter(isPlaced), [venues]);
+  const hiddenVenues = useMemo(() => venues?.filter(isHidden), [venues]);
+  const defaultValues = useMemo(() => editPlacementCastSchema.cast(venue), [
+    venue,
+  ]);
+
+  const { watch, register, reset, errors, ...rest } = useForm<FormValues>({
+    mode: "onSubmit",
+    reValidateMode: "onChange",
+    validationSchema: editPlacementSchema,
+    validationContext: {
+      template: venue?.template,
+      editing: !!venue,
+    },
+    defaultValues,
+  });
+
+  // defaultValues gets cached from first render. Need to call reset to give the latest defaults
+  useEffect(() => {
+    reset(defaultValues);
+  }, [reset, defaultValues]);
+
+  const { user } = useUser();
+  const values = watch();
+
+  const [formError, setFormError] = useState(false);
+
+  //register the icon position data
+  useEffect(() => {
+    register("placement");
+    register("width");
+    register("height");
+  }, [register]);
+
+  const onSubmit = useCallback(
+    async (vals: Partial<FormValues>) => {
+      if (!user || !venue) return;
+      try {
+        // unfortunately the typing is off for react-hook-forms.
+        await updatePlacement(vals as PlacementInput, venue.id, user);
+        setVenueId(undefined);
+      } catch (e) {
+        setFormError(true);
+        console.error(e);
+      }
+    },
+    [user, venue]
+  );
+
+  const mapIconUrl = useMemo(() => {
+    const file = values.mapIconImageFile;
+    if (file && file.length > 0) return URL.createObjectURL(file[0]);
+    return values.mapIconImageUrl;
+  }, [values.mapIconImageFile, values.mapIconImageUrl]);
+  const iconsMap = useMemo(
+    () =>
+      mapIconUrl
+        ? {
+            [iconPositionFieldName]: {
+              width: defaultValues?.width ?? PLAYA_VENUE_SIZE,
+              height: defaultValues?.height ?? PLAYA_VENUE_SIZE,
+              top: defaultValues?.placement?.y ?? 0,
+              left: defaultValues?.placement?.x ?? 0,
+              url: mapIconUrl,
+            },
+          }
+        : undefined,
+    [mapIconUrl, defaultValues]
+  );
+
+  return (
+    <>
+      <div className="page-container-adminpanel-placement">
+        {venue && (
+          <div className="venue-preview">
+            <h4 className="heading">Edit venue/room links</h4>
+            <div style={{ marginBottom: 10 }}>
+              <b>Selected Venue:</b> {venue.name}
+            </div>
+            <Link
+              className="btn btn-primary"
+              target="_blank"
+              to={`/admin/venue/edit/${venue.id}`}
+            >
+              Edit Venue
+            </Link>
+            {isCampVenue(venue) && (
```

Comment:
This should check if its camp or partymap, something like this:
`
HAS_ROOMS_TEMPLATES.includes(venue.template)
`